tests/evaluation/metrics_test.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+
+import numpy as np
+
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+
+class TestGtsfmMetric(unittest.TestCase):
+    """Unit tests for GtsfmMetric class."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self._metric_dict_quartiles = {
+            "foo_metric": {
+                "summary": {
+                    "min": 0,
+                    "max": 7,
+                    "median": 3.5,
+                    "mean": 3.5,
+                    "stddev": 2,
+                    "quartiles": {
+                        "q0": 0,
+                        "q1": 1,
+                        "q2": 3,
+                        "q3": 5,
+                        "q4": 7,
+                    },
+                },
+                "full_data": [0, 1, 2, 3, 4, 5, 6, 7],
+            }
+        }
+        self._metric_dict_histogram = {
+            "bar_metric": {
+                "summary": {
+                    "min": 0,
+                    "max": 3,
+                    "median": 1.5,
+                    "mean": 1.5,
+                    "stddev": 2,
+                    "histogram": {
+                        "0": 2,
+                        "1": 2,
+                        "2": 2,
+                        "3": 2,
+                    },
+                },
+                "full_data": [0, 1, 0, 2, 2, 1, 3, 3],
+            }
+        }
+        self._metric_dict_no_data = copy.deepcopy(self._metric_dict_histogram)
+        del self._metric_dict_no_data["bar_metric"]["full_data"]
+
+    def test_create_scalar_metric(self) -> None:
+        metric = GtsfmMetric("a_scalar", 2)
+        self.assertEqual(metric.name, "a_scalar")
+        np.testing.assert_equal(metric.data, np.array([2]))
+        self.assertEqual(metric.plot_type, GtsfmMetric.PlotType.BAR)
+
+    def test_create_1d_distribution_metric(self) -> None:
+        data = np.array([1, 2, 3, 4, 5, 6], dtype=np.float32)
+        metric = GtsfmMetric("dist_metric", data)
+        self.assertEqual(metric.name, "dist_metric")
+        np.testing.assert_equal(metric.data, data)
+        self.assertEqual(metric.plot_type, GtsfmMetric.PlotType.BOX)
+
+    def test_parses_from_dict_scalar(self) -> None:
+        scalar_metric_dict = {"foo_metric": 2}
+        parsed_metric = GtsfmMetric.parse_from_dict(scalar_metric_dict)
+        self.assertEqual(parsed_metric.name, "foo_metric")
+        np.testing.assert_equal(parsed_metric.data, 2)
+
+    def test_parses_from_dict_1D_distribution(self) -> None:
+        parsed_metric = GtsfmMetric.parse_from_dict(self._metric_dict_quartiles)
+        self.assertEqual(parsed_metric.name, "foo_metric")
+        self.assertEqual(parsed_metric.plot_type, GtsfmMetric.PlotType.BOX)
+        self.assertIn("quartiles", parsed_metric.summary)
+        self.assertIn("full_data", parsed_metric.get_metric_as_dict()[parsed_metric.name])
+
+    def test_parses_from_dict_1D_distribution_histogram(self) -> None:
```

Comment:
simple 1-line docstrings would make this even easier to follow, for each unit test

Reply:
thanks for adding these